File: Array/Python/_0283_MoveZeroes.py
from typing import List


class Solution:
    # Iteration
    # Time Complexity: BigO(N)
    # Space Complexity: BigO(N)
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index = 0
        for i in nums:
            if i != 0:
                nums[index] = i
                index += 1

        for i in range(index, len(nums)):
            nums[i] = 0

File: Array/TypeScript/_0283_MoveZeroes.ts
/**
 * Do not return anything, modify nums in-place instead.
 * Iteration
 * Time Complexity: BigO(N)
 * Space Complexity: BigO(N)
 */
function moveZeroes(nums: number[]): void {
  let zeros = nums.filter((num) => num == 0).length;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == 0 && zeros != 0) {
      let zero = nums.splice(i, 1)[0];
      nums.push(zero);
      zeros -= 1;
      i -= 1;
    }
  }
}
